src/libraries/System.Private.CoreLib/src/System/Numerics/Decimal128.cs

```diff
@@ -0,0 +1,268 @@
+            return Number.ParseDecimal128(s, style, NumberFormatInfo.GetInstance(provider));
+        }
+        public static Decimal128 Parse(string s, NumberStyles style, IFormatProvider? provider)
+        {
+            if (s is null)
+            {
+                ThrowHelper.ThrowArgumentNullException(ExceptionArgument.s);
+            }
+            return Parse(s.AsSpan(), style, provider);
+        }
+
+        public static bool TryParse([NotNullWhen(true)] string? s, out Decimal128 result) => TryParse(s, NumberStyles.Number, provider: null, out result);
+        public static bool TryParse(ReadOnlySpan<char> s, out Decimal128 result) => TryParse(s, NumberStyles.Number, provider: null, out result);
+        public static bool TryParse(ReadOnlySpan<char> s, IFormatProvider? provider, [MaybeNullWhen(false)] out Decimal128 result) => TryParse(s, NumberStyles.Number, provider, out result);
+        public static bool TryParse([NotNullWhen(true)] string? s, IFormatProvider? provider, [MaybeNullWhen(false)] out Decimal128 result) => TryParse(s, NumberStyles.Number, provider, out result);
+        public static bool TryParse(ReadOnlySpan<char> s, NumberStyles style, IFormatProvider? provider, [MaybeNullWhen(false)] out Decimal128 result)
+        {
+            NumberFormatInfo.ValidateParseStyleFloatingPoint(style);
+            return Number.TryParseDecimalIeee754<Decimal128, Int128, char>(s, style, NumberFormatInfo.GetInstance(provider), out result) == Number.ParsingStatus.OK;
+        }
+        public static bool TryParse([NotNullWhen(true)] string? s, NumberStyles style, IFormatProvider? provider, [MaybeNullWhen(false)] out Decimal128 result)
+        {
+            NumberFormatInfo.ValidateParseStyleFloatingPoint(style);
+
+            if (s == null)
+            {
+                result = default;
+                return false;
+            }
+            return Number.TryParseDecimalIeee754<Decimal128, Int128, char>(s.AsSpan(), style, NumberFormatInfo.GetInstance(provider), out result) == Number.ParsingStatus.OK;
+        }
+
+        // Compares this object to another object, returning an integer that
+        // indicates the relationship.
+        // Returns a value less than zero if this  object
+        // null is considered to be less than any instance.
+        // If object is not of type Decimal128, this method throws an ArgumentException.
+        //
+        public int CompareTo(object? value)
+        {
+            if (value == null)
+            {
+                return 1;
+            }
+
+            if (value is not Decimal128 i)
+            {
+                throw new ArgumentException(SR.Arg_MustBeDecimal128);
+            }
+
+            var current = new UInt128(_upper, _lower);
+            var other = new UInt128(i._upper, i._lower);
+
+            return Number.CompareDecimalIeee754<Decimal128, Int128, UInt128>(current, other);
+        }
```

Comment:
nit: Typically you would use `other` instead of `i` as it is much more meaningful.

But also this would typically be simplified to:
```csharp
if (value is not Decimal128 other)
{
    return (value is null) ? 1 : throw new ArgumentException(SR.Arg_MustBeDecimal128);
}
return CompareTo(other);
```

As this minimizes code duplication and helps keep the logic clearer/more readable.